File: app/src/main/java/com/aavens/mindloft/ui/detail/ThingTypeDialogFragment.java
package com.aavens.mindloft.ui.detail;

import android.app.AlertDialog;
import android.app.Dialog;
import android.app.DialogFragment;
import android.content.Context;
import android.content.DialogInterface;
import android.os.Bundle;

import com.aavens.mindloft.R;
import com.aavens.mindloft.listeners.ThingsTypeDialogListener;

public class ThingTypeDialogFragment extends DialogFragment {

    private static final String INDEX_ARG = "INDEX_ARG";

    private ThingsTypeDialogListener listener;

    public static ThingTypeDialogFragment newInstance(ThingsTypeDialogListener listener) {
        ThingTypeDialogFragment fragment = new ThingTypeDialogFragment();
        fragment.listener = listener;
        return fragment;
    }

    public ThingTypeDialogFragment() {
    }

    @Override
    public Dialog onCreateDialog(Bundle savedInstanceState) {
        AlertDialog.Builder builder = new AlertDialog.Builder(getActivity());
        builder.setTitle(R.string.choose_dialog_title);
        builder.setItems(R.array.thing_types_array, new DialogInterface.OnClickListener() {
            @Override
            public void onClick(DialogInterface dialog, int which) {
                listener.onThingTypeInteraction(which);
            }
        });
        return builder.create();
    }

    @Override
    public void onAttach(Context context) {
        super.onAttach(context);
        if (context instanceof ThingsTypeDialogListener) {
            listener = (ThingsTypeDialogListener) context;
        } else {
            throw new RuntimeException(context.toString()
                    + " must implement OnFragmentInteractionListener");
        }
    }

    @Override
    public void onDetach() {
        super.onDetach();
        listener = null;
    }
}

File: app/src/main/java/com/aavens/mindloft/ui/detail/ThingTypeDialogFragment.kt
package com.aavens.mindloft.ui.detail

import android.app.AlertDialog
import android.app.Dialog
import android.app.DialogFragment
import android.content.Context
import android.os.Bundle

import com.aavens.mindloft.R
import com.aavens.mindloft.listeners.ThingsTypeDialogListener

class ThingTypeDialogFragment: DialogFragment() {

    private var listener: ThingsTypeDialogListener? = null

    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
        val builder = AlertDialog.Builder(activity)
        builder.setTitle(R.string.choose_dialog_title)
        builder.setItems(R.array.thing_types_array) { dialog, which -> listener?.onThingTypeInteraction(which) }
        return builder.create()
    }

    override fun onAttach(context: Context) {
        super.onAttach(context)
        if (context is ThingsTypeDialogListener) {
            listener = context
        } else {
            throw RuntimeException(context.toString() + " must implement OnFragmentInteractionListener")
        }
    }

    override fun onDetach() {
        super.onDetach()
        listener = null
    }

    companion object {

        private val INDEX_ARG = "INDEX_ARG"

        fun newInstance(listener: ThingsTypeDialogListener): ThingTypeDialogFragment {
            val fragment = ThingTypeDialogFragment()
            fragment.listener = listener
            return fragment
        }
    }
}
